Src/bootloader.c

```diff
@@ -135,7 +135,7 @@ void BL_ReadMemory_Command(uint32_t address, uint8_t nob, uint8_t *pData)
   uint8_t addr_frame[5];
   uint8_t nob_frame[2];
   uint8_t dummy = 0x00U;
-
+  uint16_t nobrx = nob+1;

```

Comment:
Yes - agreed - it would be easier to understand and the fact that the bootloader uses nob - 1 internally can be hidden away as implementation detail